extension/beatsauthextension/authenticator.go

```diff
@@ -77,13 +86,22 @@ func (a *authenticator) RoundTripper(base http.RoundTripper) (http.RoundTripper,
 }
 
 func (a *authenticator) configureTransport(transport *http.Transport) error {
+
 	if a.tlsConfig != nil {
-		// injecting verifyConnection here, keeping all other fields on TLSClientConfig intact
-		beatTLSConfig := a.tlsConfig.BuildModuleClientConfig(a.cfg.TLS.ServerName)
+		a.mx.Lock()
+		defer a.mx.Unlock()
+
+		// copy incoming CertPool into our tls config
+		// because ca_trusted_fingerprint will be appended to CertPool
+		a.tlsConfig.RootCAs = transport.TLSClientConfig.RootCAs
```

Comment:
```suggestion
		// copy incoming CertPool into our tls config
		// because ca_trusted_fingerprint will be appended to CertPool
		tlsConfig := *a.tlsConfig // copy before updating, configureTransport may be called concurrently
		tlsConfig.RootCAs = transport.TLSClientConfig.RootCAs
		beatTLSConfig := tlsConfig.BuildModuleClientConfig(transport.TLSClientConfig.ServerName)
```

I _think_ what you did works but it seems unnecessarily complicated to use a mutex. Can we make a copy of `a.tlsConfig` instead?